### ISSUE TITLE
Unhandled 'An error occured waiting for the image to become available' in ansible_module_ec2_ami_copy.py

### ISSUE THREAD

**Issue by piotrkarbowski-ciklum** [ansible/ansible#26971]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
`ansible_module_ec2_ami_copy.py`

##### ANSIBLE VERSION
```
ansible==2.3.1.0 
```

##### OS / ENVIRONMENT

Linux, AWS CodeBuild

##### SUMMARY

Today, while executing a task with `ec2_ami_copy:` ansible execution crashed with:

    An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'WaiterError' object has no attribute 'reason'
    /aws/codebuild/core-build-ami c6b4c0b1-5035-4833-8456-f486d8f78e27 fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):
      File \"/tmp/ansible_wM5kH0/ansible_module_ec2_ami_copy.py\", line 226, in <module>
        main()
      File \"/tmp/ansible_wM5kH0/ansible_module_ec2_ami_copy.py\", line 222, in main
    copy_image(module, ec2)
      File \"/tmp/ansible_wM5kH0/ansible_module_ec2_ami_copy.py\", line 183, in copy_image
    module.fail_json(msg='An error occured waiting for the image to become available. (%s)' %  we.reason)
    AttributeError: 'WaiterError' object has no attribute 'reason'
    ", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}

##### STEPS TO REPRODUCE

Occasional 

```yaml
  - name: copy image
    ec2_ami_copy:
      name: tmp-encrypted-{{ ubuntu_ami_result.results[0].name }}-{{ ansible_date_time.iso8601_basic_short }}
      source_region: "{{ aws_region }}"
      region: "{{ aws_region }}"
      source_image_id: "{{ ubuntu_ami_result.results[0].ami_id }}"
      encrypted: yes
      wait: yes
      tags:
        terminate_after: 1h
    register: image_id
```

##### EXPECTED RESULTS

Going forward to next task.

##### ACTUAL RESULTS

Not handled error resulting in nasty crash

CC @adamchainz

**Comment by adamchainz** [comment]
Looks like it should just be `.message` instead of `.reason`:

```python
In [1]: from botocore.exceptions import ClientError, NoCredentialsError, NoRegionError, WaiterError

In [2]: WaiterError??
Init signature: WaiterError(self, name, reason, last_response)
Source:
class WaiterError(BotoCoreError):
    """Waiter failed to reach desired state."""
    fmt = 'Waiter {name} failed: {reason}'

    def __init__(self, name, reason, last_response):
        super(WaiterError, self).__init__(name=name, reason=reason)
        self.last_response = last_response
File:           /usr/local/lib/python2.7/site-packages/botocore/exceptions.py
Type:           type

In [3]: WaiterError('a', 'a', {})
Out[3]: botocore.exceptions.WaiterError(u'Waiter a failed: a')

In [4]: WaiterError('a', 'a', {}).reason
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-3b104648e88b> in <module>()
----> 1 WaiterError('a', 'a', {}).reason

AttributeError: 'WaiterError' object has no attribute 'reason'

In [5]: dir(WaiterError('a', 'a', {}))
Out[5]:
['__class__',
 '__delattr__',
 '__dict__',
 '__doc__',
 '__format__',
 '__getattribute__',
 '__getitem__',
 '__getslice__',
 '__hash__',
 '__init__',
 '__module__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__setstate__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 '__unicode__',
 '__weakref__',
 'args',
 'fmt',
 'kwargs',
 'last_response',
 'message']

In [6]: WaiterError('a', 'a', {}).message
/usr/local/bin/ipython:1: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  #!/usr/local/opt/python/bin/python2.7
Out[6]: u'Waiter a failed: a'
```

**Comment by adamchainz** [comment]
Or in fact to avoid deprecation warning, `str(we)`:

```python
In [7]: str(WaiterError('a', 'a', {}))
Out[7]: 'Waiter a failed: a'
```

**Comment by s-hertel** [comment]
@piotrkarbowski-ciklum or @adamchainz Want to make a pull request to fix this?

**Comment by adamchainz** [comment]
I asked @piotrkarbowski-ciklum to but he hasn't got around to it

**Comment by adamchainz** [comment]
(we work in the same team)

**Comment by s-hertel** [comment]
Fixed by a260d99. Thank you @piotrkarbowski-ciklum @adamchainz!
